src/centrality/pagerank.jl

```diff
@@ -18,31 +18,30 @@ function pagerank(
     n=100::Integer, 
     ϵ=1.0e-6
     ) where U <: Integer
-    out_degree = [outdegree(g, u) for u in vertices(g)]
-    # collect dangling nodes
-    dangling_nodes = [v for v in vertices(g) if out_degree[v] == 0]
+    α_div_outdegree = Vector{Float64}(undef,nv(g))
+    dangling_nodes = Vector{U}()
+    for v in vertices(g)
+        outdegree(g, v) == 0 && push!(dangling_nodes, v)
```

Comment:
I think LightGraphs convention does not short-circuit mutation statements like this.
cc: @sbromberger